event/UserAttendance.tsx

```diff
@@ -0,0 +1,101 @@
+import { ReactNode, createContext, useContext } from "react"
+import { StyleProp, ViewStyle } from "react-native"
+import {
+  JoinEventButton,
+  JoinEventPermission,
+  JoinEventPermissionsSheetView,
+  JoinEventRequest,
+  JoinEventResult,
+  joinEvent,
+  loadJoinEventPermissions,
+  useJoinEvent
+} from "./JoinEvent"
+import {
+  LeaveEventButton,
+  LeaveEventResult,
+  leaveEvent,
+  useLeaveEvent
+} from "./LeaveEvent"
+import { EventID, isAttendingEvent } from "TiFShared/domain-models/Event"
+import { TiFAPI } from "TiFShared/api"
+import { mergeWithPartial } from "TiFShared/lib/Object"
+import { EventRegionMonitor } from "@arrival-tracking"
+import { TrueRegionMonitor } from "@arrival-tracking/region-monitoring/MockRegionMonitors"
+import { ClientSideEvent } from "./ClientSideEvent"
+
+export type EventUserAttendanceContextValues = {
+  monitor: EventRegionMonitor
```

Comment:
What does monitor perform here? Little confused

Reply:
It's needed for `useJoinEvent`